src/txmempool.h

```diff
@@ -375,6 +375,7 @@ class CTxMemPool
     void removeForBlock(const std::vector<CTransaction>& vtx, unsigned int nBlockHeight,
                         std::list<CTransaction>& conflicts, bool fCurrentEstimate = true);
     void clear();
```

Comment:
Usually projects have some convention for this (eg _XX and XX, or XX_nolock and XX) . Where one is used internally (when lock already held) and the other externally (locks and calls inner function). 
Preferable to have the names as similar as possible with just the lock/nonlock difference. `lockAndClear` and `clearSync` are too far apart to be easily recognizable as a set. So IMO the current naming is better.